static/js/components/forms/SiteContentForm.test.tsx

```diff
@@ -1,239 +1,236 @@
-    onSubmitStub: SinonStub,
-    configItem: EditableConfigItem,
-    content: WebsiteContent,
-    mockValidationSchema: FormSchema,
-    website: Website,
-    setDirtyStub: SinonStub
-
-  beforeEach(() => {
-    sandbox = sinon.createSandbox()
-    content = makeWebsiteContentDetail()
-    content.content_context = times(() => makeWebsiteContentDetail(), 3)
-    configItem = makeEditableConfigItem(content.type)
-    // @ts-ignore
-    renameNestedFields.mockImplementation(() => [])
-    website = makeWebsiteDetail()
-    // @ts-ignore
-    useWebsite.mockReturnValue(website)
-    onSubmitStub = sandbox.stub()
-    setDirtyStub = sandbox.stub()
-  })
+function setupData() {
+  const content = makeWebsiteContentDetail()
+  content.content_context = times(() => makeWebsiteContentDetail(), 3)
+  const configItem = makeEditableConfigItem(content.type)
+  const website = makeWebsiteDetail()
+  // @ts-ignore
+  useWebsite.mockReturnValue(website)
+  const onSubmit = jest.fn()
+  const setDirty = jest.fn()
+  const editorState: WebsiteContentModalState = createModalState("adding")
+
+  return {
+    content,
+    configItem,
+    onSubmit,
+    setDirty,
+    editorState,
+    website
+  }
+}
 
-  afterEach(() => {
-    sandbox.restore()
-  })
+function setup(props: Partial<FormProps> = {}) {
+  const data = setupData()
 
-  const renderForm = (props = {}) =>
-    shallow(
-      <SiteContentForm
-        configItem={configItem}
-        content={content}
-        onSubmit={onSubmitStub}
-        editorState={createModalState("adding")}
-        setDirty={setDirtyStub}
-        {...props}
-      />
-    )
+  const form = mount(<SiteContentForm {...data} {...props} />)
 
-  const renderInnerForm = (
-    editorState: WebsiteContentModalState,
-    formikChildProps: { [key: string]: any } = {}
-  ) => {
-    const wrapper = renderForm({ editorState })
-    return (
-      wrapper
-        .find("Formik")
-      // @ts-ignore
-        .renderProp("children")({
-          ...defaultFormikChildProps,
-          ...formikChildProps
+  return {
+    form,
+    ...data
+  }
+}
+
+/**
+ * A separate setup function for inserting specific test values
+ * into the inner form component
+ */
+function setupInnerForm(props: Partial<InnerFormProps> = {}) {
+  const data = setupData()
+
+  const form = mount(
+    <Formik
+      onSubmit={data.onSubmit}
+      validate={jest.fn()}
+      initialValues={{}}
+      enableReinitialize={true}
+    >
+      {formikProps => <FormFields {...formikProps} {...data} {...props} />}
+    </Formik>
+  )
+
+  return { form, ...data }
+}
+
+const EDITOR_STATES: WebsiteContentModalState[] = [
+  createModalState("adding"),
+  createModalState("editing", "id")
+]
+
+type ESBoolMatrix = [WebsiteContentModalState, boolean][]
+
+const EDITOR_STATE_BOOLEAN_MATRIX: ESBoolMatrix = EDITOR_STATES.map(
+  editorState => [
+    [editorState, true],
+    [editorState, false]
+  ]
+).flat() as any // :/ unfortunate but necessary
```

Comment:
You can use `as ESBoolMatrix` here instead of `as any` if you want... `as ESBoolMatrix` still does some checking, e.g. `5 as ESBoolMatrix` will error.

Reply:
mmm will change that, I thought I tried it but I must not have